lib/src/objects/config_object.cc

```diff
@@ -51,4 +52,42 @@ namespace hocon {
         return config_value_type::OBJECT;
     }
 
+    shared_ptr<const config_mergeable> config_object::with_fallback(shared_ptr<const config_mergeable> mergeable) const {
+        return dynamic_pointer_cast<const config_value>(config_value::with_fallback(mergeable));
+    }
+
+    shared_origin config_object::merge_origins(std::vector<shared_value> const& stack) {
+        if (stack.empty()) {
+            throw config_exception("can't merge origins on empty list");
+        }
+
+        vector<shared_origin> origins;
+        shared_origin first_origin = nullptr;
+        int num_merged = 0;
+
+        for (shared_value v : stack) {
+            if (first_origin == nullptr) {
+                first_origin = v->origin();
+            }
+
+            auto cv = dynamic_pointer_cast<const config_object>(v);
+            if (cv && cv->get_resolve_status() == resolve_status::RESOLVED && cv->is_empty()) {
+                // don't include empty files or the .empty()
+                // config in the description, since they are
+                // likely to be "implementation details"
+            } else {
+                origins.push_back(v->origin());
+                num_merged += 1;
```

Comment:
I know it mirrors the original, but origins.size() will always be equal to num_merged. We could do away with tracking it separately and check `origins.empty()` at line 84.